File: backend-node/src/entity/Plan.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity('Plan')
export default class Plan{
    @PrimaryGeneratedColumn()
    public id!: number;

    @Column("varchar", { length: 100, nullable: false })
    private name: string;

    @Column("int", { nullable: false })
    private days?: number;

    @Column("float", { nullable: false })
    private price: number;

    @Column("text", { nullable: false })
    private description: string;


    constructor(name: string, price: number, description: string, days: number){
        this.name = name;
        this.price = price;
        this.description = description;
        this.days = days;
    }

    public getId(): number {
        return this.id;
    }

    public getName(): string {
        return this.name;
    }

    public getPrice(): number {
        return this.price;
    }

    public getDays(): number {
        return this.days!;
    }

    public getDescription(): string {
        return this.description;
    }
}

File: backend-java/src/main/java/kafka/system/br/AutoFrota/model/Plan.java
package kafka.system.br.AutoFrota.model;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Table(name = "plan")
@Entity(name = "Plan")
public class Plan {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "days", nullable = false)
    private int days;

    @Column(name = "price", nullable = false)
    private double price;

    @Column(name = "amount_vehicles", nullable = false)
    private int amountVehicles;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "feature_id")
    private Feature feature;

    public Plan() {
    }

    public Plan(Long id, String name, int days, double price, int amountVehicles, Feature feature) {
        this.id = id;
        this.name = name;
        this.days = days;
        this.price = price;
        this.amountVehicles = amountVehicles;
        this.feature = feature;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getDays() {
        return days;
    }

    public void setDays(int days) {
        this.days = days;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getAmountVehicles() {
        return amountVehicles;
    }

    public void setAmountVehicles(int amountVehicles) {
        this.amountVehicles = amountVehicles;
    }

    public Feature getFeature() {
        return feature;
    }

    public void setFeature(Feature feature) {
        this.feature = feature;
    }
}
